src/arithmetic/arithmetic_expression.h

```diff
@@ -31,13 +31,17 @@ typedef enum {
 	AR_OP_FUNC,
 } AR_OPType;
 
-/* AR_OperandNodeType type of leaf node,
- * either a constant: 3, a variable: node.property, or a parameter: $p. */
+/* AR_OperandNodeType - Type of an expression tree's leaf node. Can be:
+ * A constant, e.g. 3
+ * A variable, e.g. node.property
+ * A parameter, e.g. $p.
+ * A directive to store the current record. */
 typedef enum {
 	AR_EXP_OP_UNKNOWN,
 	AR_EXP_CONSTANT,
 	AR_EXP_VARIADIC,
-	AR_EXP_PARAM
+	AR_EXP_PARAM,
+	AR_EXP_BORROW_RECORD
 } AR_OperandNodeType;
```

Comment:
Let move the above comment into the `enum`
`AR_EXP_CONSTANT,    //  A constant, e.g. 3`